torchaudio/functional.py

```diff
@@ -218,42 +217,52 @@ def istft(
 def spectrogram(
     waveform, pad, window, n_fft, hop_length, win_length, power, normalized
 ):
-    # type: (Tensor, int, Tensor, int, int, int, int, bool) -> Tensor
+    # type: (Tensor, int, Tensor, int, int, int, Optional[int], bool) -> Tensor
     r"""
     spectrogram(waveform, pad, window, n_fft, hop_length, win_length, power, normalized)
 
-    Create a spectrogram from a raw audio signal.
+    Create a spectrogram or a batch of spectrograms from a raw audio signal.
+    The spectrogram can be either magnitude-only or complex.
 
     Args:
-        waveform (torch.Tensor): Tensor of audio of dimension (channel, time)
+        waveform (torch.Tensor): Tensor of audio of dimension (*, channel, time)
```

Comment:
I'd explicitly call this 'batch', i.e. ([batch], channel, time) or such.

Reply:
This is more general than batch. It works with a tensor of any dimension ending in (channel, time). In fact, this could be updated to (*, time) by making channel optional. Saying "(batch, channel, time) or (channel, time) or (time)" is a little heavy. And saying "([batch], [channel], time)" hides the generality. Thoughts?

Reply:
I can't think of anything more compact. Just by following what it is something like `..(*, time), e.g., (time), (channel, time), (batch, channel, time)`? It's kinda verbose but if we can reduce its ambiguity :)

Reply:
I agree with reducing ambiguity :)

Is `(..., time)` clearer than `(*, time)`?

Reply:
Hmm............ maybe? (But if I might have confused you, I meant `..audio of dimensions (*, time), e.g., (time), blah -`.) The current `torch.stft` doc uses `(*, time, ..)` to indicate an optional batch dimension which is more limited than we want here. Probably `(..., time)` makes more sense then.

Reply:
Good point, the better globbing to use would be `(**, time)` not `(*, time)` for `(..., time)` :)

I do not recommend using that though :)